src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -71,58 +68,149 @@ class F extends ValidationTest {
-    { width: 32, height: 32, mipLevelCount: 1, _success: true }, // mipLevelCount of 1 is allowed
-    { width: 32, height: 32, mipLevelCount: 3, _success: true }, // non full mip chains are allowed (Mip leve sized: 32, 16, 8)
-    { width: 32, height: 32, mipLevelCount: 6, _success: true }, // full mip chains are allowed (Mip level sizes: 32, 16, 8, 4, 2, 1)
-    { width: 31, height: 32, mipLevelCount: 6, _success: true }, // full mip chains are allowed (Mip level sizes: 31x32, 15x16, 7x8, 3x4, 1x2, 1x1)
-    { width: 32, height: 31, mipLevelCount: 6, _success: true }, // full mip chains are allowed (Mip level sizes: 32x31, 16x15, 8x7, 4x3, 2x1, 1x1)
-    { width: 31, height: 32, mipLevelCount: 7, _success: false }, // too big mip chains on width are disallowed (Mip level sizes: 31x32, 15x16, 7x8, 3x4, 1x2, 1x1, ?x?)
-    { width: 32, height: 31, mipLevelCount: 7, _success: false }, // too big mip chains on height are disallowed (Mip level sizes: 32x31, 16x15, 8x7, 4x3, 2x1, 1x1, ?x?)
-    { width: 32, height: 32, mipLevelCount: 100, _success: false }, // undefined shift check if miplevel is bigger than the integer bit width
-    { width: 32, height: 8, mipLevelCount: 6, _success: true }, // non square mip map halves the resolution until a 1x1 dimension. (Mip maps: 32 * 8, 16 * 4, 8 * 2, 4 * 1, 2 * 1, 1 * 1)
-    { width: 32, height: 32, arrayLayerCount: 64, mipLevelCount: 7, _success: false }, // array layer count for 2D texture should not be taken account to calculate mip levels. (Mip maps: 32 * 32 * 64, 16 * 16 * 64, 8 * 8 * 64, 4 * 4 * 64, 2 * 2 * 64, 1 * 1 * 64)
+    { size: [32, 32], mipLevelCount: 6, _success: true }, // full mip chains for 2D texture are allowed (Mip level sizes: 32x32, 16x32, 8x8, 4x4, 2x2, 1x1)
+    { size: [31, 32], mipLevelCount: 6, _success: true }, // full mip chains for 2D texture are allowed (Mip level sizes: 31x32, 15x16, 7x8, 3x4, 1x2, 1x1)
+    { size: [32, 31], mipLevelCount: 6, _success: true }, // full mip chains for 2D texture are allowed (Mip level sizes: 32x31, 16x15, 8x7, 4x3, 2x1, 1x1)
+    { size: [31, 32], mipLevelCount: 7, _success: false }, // too big mip chains on width for 2D texture are disallowed (Mip level sizes: 31x32, 15x16, 7x8, 3x4, 1x2, 1x1, ?x?)
+    { size: [32, 31], mipLevelCount: 7, _success: false }, // too big mip chains on height for 2D texture are disallowed (Mip level sizes: 32x31, 16x15, 8x7, 4x3, 2x1, 1x1, ?x?)
+    { size: [31, 31], mipLevelCount: 5, _success: true }, // full mip chains for non-power-of-two 2D texture are allowed (Mip level sizes: 31x31, 15x15, 7x7, 3x3, 1x1)
+    { size: [31, 31], mipLevelCount: 6, _success: false }, // too big mip chains for non-power-of-two 2D texture are disallowed (Mip level sizes: 31x31, 15x15, 7x7, 3x3, 1x1, ?x?)
+    { size: [32, 1, 1], dimension: '1d' as const, mipLevelCount: 6, _success: true }, // full mip chains for 1D texture are allowed (Mip level sizes: 32, 16, 8, 4, 2, 1)
```

Comment:
uNit: These could be just `size: [32]`

Reply:
Done. But when I run the test in latest Chromium, it complains that size's length should be 3. Is there anything wrong in the test (or in Chromium/Dawn)?

Reply:
This restriction should have been lifted in https://chromium-review.googlesource.com/c/chromium/src/+/2478976, but maybe we missed something, can you check and file an issue if we did?

Reply:
Done. see the crbug at https://bugs.chromium.org/p/chromium/issues/detail?id=1180301